tests/tensorflow/test_printing.py

```diff
@@ -79,3 +79,47 @@ def test_unknown_shape():
     expected_out = 'Tensor(Placeholder):0,\tshape=Unknown\t"A:0"\n'
 
     assert std_out.getvalue() == expected_out.lstrip()
+
+
+@pytest.mark.usefixtures("run_with_tensorflow")
+@run_in_graph_mode
+def test_shape_index():
+    """Make sure graph indexing functions as expected."""
+
+    A = tf.compat.v1.placeholder('float', name='A',
+                                 shape=tf.TensorShape([None, None]))
+    x = tf.compat.v1.placeholder('float', name='x',
+                                 shape=tf.TensorShape([None, 1]))
+    y = tf.multiply(1.0, x, name='y')
+
+    z = tf.matmul(A, tf.add(y, y, name='x_p_y'), name='A_dot')
+
+    range_out = io.StringIO()
+    with redirect_stdout(range_out):        
+        tf_dprint(z, depth_lower=1, depth_upper=4)
+
+    expected_out = textwrap.dedent('''
+    ... Op(MatMul)\t"A_dot"
+    ... |  Tensor(Placeholder):0,\tshape=[None, None]\t"A:0"
+    ... |  Tensor(Add):0,\tshape=[None, 1]\t"x_p_y:0"
+    ... |  |  Op(Add)\t"x_p_y"
+    ''')
```

Comment:
Shouldn't this end with a `... |  |  |  ...` to indicate the truncated subgraph?

Reply:
Added in latest commit.